mlrun/api/api/endpoints/projects.py

```diff
@@ -322,6 +324,98 @@ async def get_project_summary(
     return project_summary
 
 
+@router.post("/projects/{name}/load", response_model=mlrun.api.schemas.BackgroundTask)
+async def load_project(
+    name: str,
+    url: str,
+    background_tasks: fastapi.BackgroundTasks,
+    auth_info: mlrun.api.schemas.AuthInfo = fastapi.Depends(
+        mlrun.api.api.deps.authenticate_request
+    ),
+    db_session: sqlalchemy.orm.Session = fastapi.Depends(
+        mlrun.api.api.deps.get_db_session
+    ),
+):
+    """
+    Loading a project remotely from a given source.
+
+    :param name:                project name
+    :param url:                 git or tar.gz or .zip sources archive path e.g.:
+                                git://github.com/mlrun/demo-xgb-project.git
+                                http://mysite/archived-project.zip
+                                The git project should include the project yaml file.
+    :param background_tasks:    injected automatically by fastapi
+    :param auth_info:           auth info of the request
+    :param db_session:          session that manages the current dialog with the database
+
+    :returns: a BackgroundTask object, with details on execution process and its status
+    """
+
+    def _run_load_project(
+        runner: mlrun.run.KubejobRuntime,
+        project_: mlrun.api.schemas.Project,
+    ):
+        """
+        Wrapper function to load project for running as a background task
+
+        :param runner:  load project function runner object
+        :param project_: MLRun project
+
+        :returns: run context object (RunObject) with run metadata, results and status
+        """
+        return mlrun.api.crud.WorkflowRunners().run(
+            runner=runner,
+            project=project_,
+            load_only=True,
+        )
+
+    project = mlrun.api.schemas.Project(
+        metadata=mlrun.api.schemas.ProjectMetadata(name=name),
+        spec=mlrun.api.schemas.ProjectSpec(source=url),
+    )
+
+    # We must create the project before we run the remote load_project function because
+    # we want this function will be running under the project itself instead of the default project.
+    project, _ = get_project_member().create_project(
```

Comment:
Note that, anything, everything under `async` that contains blocking IO operations must be running in a threadpool, or otherwise, you will block the main event loop and cause the api to hang forever.

Reply:
thanks for explaining, fixed.